.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+        default: '["ubuntu-latest"]'
+      commands:
+        required: false
+        type: string
+        default: '["hardhat compile", "hardhat test solidity"]'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-edr:
+    name: Build EDR
+    if: inputs.edr-ref
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: ${{ fromJSON(inputs.runners) }}
+    defaults:
+      run:
+        shell: bash
+        working-directory: crates/edr_napi
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: NomicFoundation/edr
+          ref: ${{ inputs.edr-ref }}
+      - id: target
+        env:
+          RUNNER: ${{ matrix.runner }}
+        run: |
+          if [[ "$RUNNER" == "ubuntu-latest" ]]; then
+            echo "target=x86_64-unknown-linux-gnu" >> $GITHUB_OUTPUT
+          elif [[ "$RUNNER" == "macos-latest" ]]; then
+            echo "target=aarch64-apple-darwin" >> $GITHUB_OUTPUT
+          elif [[ "$RUNNER" == "windows-latest" ]]; then
+            echo "target=x86_64-pc-windows-msvc" >> $GITHUB_OUTPUT
+          else
+            exit 1
+          fi
+      - uses: ./.github/actions/setup-node
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: stable
+      - name: Add Rust cross-compilation target
+        run: rustup target add ${{ steps.target.outputs.target }}
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile --prefer-offline
+      - name: Build
+        run: pnpm build --target ${{ steps.target.outputs.target }}
+      - name: Remove symbols
+        if: runner.os != 'MacOS'
+        run: strip *.node
+      - name: Remove symbols (macOS)
+        if: runner.os == 'MacOS'
+        run: strip -x *.node
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: edr-${{ matrix.runner }}
+          path: crates/edr_napi/edr.*.node
+          if-no-files-found: error
+
+  publish-edr:
+    name: Publish EDR
+    needs: [build-edr]
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: crates/edr_napi
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: NomicFoundation/edr
+          ref: ${{ inputs.edr-ref }}
+      - uses: ./.github/actions/setup-node
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile --prefer-offline
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: crates/edr_napi/artifacts
+      - name: Install sponge
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y moreutils
+      - name: Move artifacts
+        run: pnpm artifacts
+      - name: Install artifacts
+        run: |
+          for artifact in npm/*/*.node; do
+            pnpm install --save "./$(dirname $artifact)"
+          done
```

Comment:
In actual EDR release this is handled by the prepublish script. The reason why I didn't use it is because I wanted the dependencies installed as `dependencies` rather than `optionalDependencies` and I wasn't sure if prebublishing doesn't have other side effects.